contracts/Tokens/Prime/Prime.sol

```diff
@@ -0,0 +1,810 @@
+     * @param user the account address for which the prime token will be burned
+     */
+    function burn(address user) external {
+        _checkAccessAllowed("burn(address)");
+        _burn(user);
+    }
+
+    /**
+     * @notice fetch the numbers of seconds remaining for staking period to complete
+     * @param user the account address for which we are checking the remaining time
+     * @return timeRemaining the number of seconds the user needs to wait to claim prime token
+     */
+    function claimTimeRemaining(address user) external view returns (uint256) {
+        if (stakedAt[user] == 0) revert IneligibleToClaim();
+
+        uint256 totalTimeStaked = block.timestamp - stakedAt[user];
+        if (totalTimeStaked < STAKING_PERIOD) {
+            return STAKING_PERIOD - totalTimeStaked;
+        } else {
+            return 0;
+        }
+    }
+
+    /**
+     * @notice Initializes all the markets for the user when a prime token is minted
+     * @param account the account address for which markets needs to be initialized
+     */
+    function _initializeMarkets(address account) internal {
+        address[] storage _allMarkets = allMarkets;
+        for (uint i = 0; i < _allMarkets.length; i++) {
+            address market = _allMarkets[i];
+            accrueInterest(market);
+
+            interests[market][account].rewardIndex = markets[market].rewardIndex;
+
+            uint score = _calculateScore(market, account);
+            interests[market][account].score = score;
+            markets[market].sumOfMembersScore = markets[market].sumOfMembersScore + score;
+        }
+    }
+
+    /**
+     * @notice fetch the current XVS balance of user in the XVSVault
+     * @param user the account address for which markets needs to be initialized
+     * @return xvsBalance the XVS balance of user
+     */
+    function _xvsBalanceOfUser(address user) internal view returns (uint256) {
+        (uint256 xvs, , uint256 pendingWithdrawals) = IXVSVault(xvsVault).getUserInfo(
+            xvsVaultRewardToken,
+            xvsVaultPoolId,
+            user
+        );
+        return (xvs - pendingWithdrawals);
+    }
+
+    /**
+     * @notice calculate the current score of user
+     * @param market the market for which to calculate the score
+     * @param user the account for which to calculate the score
+     * @return score the score of the user
+     */
+    function _calculateScore(address market, address user) internal returns (uint256) {
+        uint256 xvsBalanceForScore = _xvsBalanceForScore(_xvsBalanceOfUser(user));
+
+        IVToken vToken = IVToken(market);
+        uint256 borrow = vToken.borrowBalanceStored(user);
+        uint256 exchangeRate = vToken.exchangeRateStored();
+        uint256 balanceOfAccount = vToken.balanceOf(user);
+        uint256 supply = (exchangeRate * balanceOfAccount) / EXP_SCALE;
+
+
+        ResilientOracleInterface oracleInterface = ResilientOracleInterface(oracle);
```

Comment:
Would it make sense to define the storage variable as `ResilientOracleInterface`, to avoid the cast here?